sshd-sftp/src/main/java/org/apache/sshd/client/subsystem/sftp/impl/AbstractSftpClient.java

```diff
@@ -783,10 +786,6 @@ public void rename(String oldPath, String newPath, Collection<CopyMode> options)
     @Override
     public int read(Handle handle, long fileOffset, byte[] dst, int dstOffset, int len, AtomicReference<Boolean> eofSignalled)
             throws IOException {
-        if (eofSignalled != null) {
-            eofSignalled.set(null);
-        }
-
```

Comment:
We have `read` which calls `checkData` which calls `checkDataResponse`.  Having the 3 methods setting `eofSignalled` to null when entering while the only place where it can be set to a non null value is later inside the `checkDataResponse` seems a bit redundant to me.  Also, I think checking the EOF value in case an exception has been thrown is meaningless, so only keeping the `eofSignalled.set(null)` from inside `checkDataResponse` sounds enough to me.